src/sentry/api/endpoints/organization_events_has_measurements.py

```diff
@@ -0,0 +1,118 @@
+from datetime import timedelta
+
+import sentry_sdk
+from django.core.cache import cache
+from django.utils import timezone
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from sentry.api.bases import NoProjects, OrganizationEventsV2EndpointBase
+from sentry.snuba import discover
+from sentry.utils.hashlib import md5_text
+
+MEASUREMENT_TYPES = {
+    "web": [
+        "measurements.fp",
+        "measurements.fcp",
+        "measurements.lcp",
+        "measurements.fid",
+        "measurements.cls",
+    ],
+    "mobile": [
+        "measurements.app_start_cold",
+        "measurements.app_start_warm",
+        "measurements.frames_total",
+        "measurements.frames_slow",
+        "measurements.frames_frozen",
+        "measurements.stall_count",
+        "measurements.stall_stall_total_time",
+        "measurements.stall_stall_longest_time",
+    ],
+}
+
+
+class EventsHasMeasurementsQuerySerializer(serializers.Serializer):
+    transaction = serializers.CharField(max_length=200)
+    type = serializers.ChoiceField(choices=list(MEASUREMENT_TYPES.keys()))
+
+    def validate(self, data):
+        # only allow one project at a time in order to cache the results
+        # for a unique transaction
+        project_ids = self.context.get("params", {}).get("project_id", [])
+        if len(project_ids) != 1:
+            raise serializers.ValidationError("Only 1 project allowed.")
+        return data
+
+
+class OrganizationEventsHasMeasurementsEndpoint(OrganizationEventsV2EndpointBase):
+    def get_snuba_params(self, request, organization, check_global_views=True):
+        params = super().get_snuba_params(
+            request, organization, check_global_views=check_global_views
+        )
+
+        # Once an transaction begins containing measurement data, it is unlikely
+        # it will stop. So it makes more sense to always query the latest data.
+        #
+        # Additionally, to account for periods of low volume, increase the range
+        # to 7 days to have a better chance of finding an example event and provide
+        # a more consistent experience.
+        now = timezone.now()
+        params["start"] = now - timedelta(days=7)
+        params["end"] = now
+
+        return params
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        with sentry_sdk.start_span(op="discover.endpoint", description="parse params"):
+            try:
+                params = self.get_snuba_params(request, organization)
```

Comment:
nit: couldn't we just do the date adjustments here instead?
```suggestion
                params = self.get_snuba_params(request, organization)
                now = timezone.now()
                params["start"] = now - timedelta(days=7)
                params["end"] = now
```